### PR TITLE
Large improvements + fixes

### PR DESCRIPTION
**Pull request by adriantran**
LeanRooster:
- Added getContent

Emoji:
- Added slightly smiling face
- Trigger content change when starting an emoji through command

RoosterCommandBar:
- Updated Office Fabric version and consume overflowMenuProps
- Added onButtonClicked property (for scenarios like telemetry)
- Changed smiley to be slightly smiling face
- Improved color picker UX with color based hover effect for performance
- Reordered buttons once again
- Updated sprites sheet

ImageManager:
- Added check for empty image file edge case
- Trigger change event after placeholder resolution

UndoWithImagePlugin:
- Fixed undo/redo logic on when going through duplicate snapshot (one snapshot for placeholder, one snapshot for placeholder resolution--same content at the end)

Sample:
- Fixed command bar hiding logic in sample to avoid intermittent overflow button
- Added sample usage of tooltip for links in the editor

### CODE REVIEW

File: packages/roosterjs-react-command-bar/lib/plugins/RoosterCommandBarPlugin.Shortcuts.ts [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+        command: RoosterCommandBarCommands.InsertLink
+    }
+];
+
+export function getCommandFromEvent(event: PluginEvent): RoosterCommandBarCommands {
+    if (event.eventType !== PluginEventType.KeyDown) {
+        return RoosterCommandBarCommands.None;
+    }
+
+    const commands = browserData.isMac ? macCommands : winCommands;
```
Comment (pranavgk) [inline packages/roosterjs-react-command-bar/lib/plugins/RoosterCommandBarPlugin.Shortcuts.ts]: Interesting way to handle meta and ctrl keys :)

Comment (adriantran) [reply packages/roosterjs-react-command-bar/lib/plugins/RoosterCommandBarPlugin.Shortcuts.ts]: it's actually refactored from DefaultShortcut; ctrl+k link adding was actually missing mac support

File: packages/roosterjs-react-command-bar/lib/components/RoosterCommandBar.tsx [hunk 2]
```diff
@@ -37,13 +37,21 @@ export default class RoosterCommandBar extends React.PureComponent<RoosterComman
...
         return (
             <div className={css("rooster-command-bar", className)}>
-                <CommandBar className={"command-bar"} items={this._buttons} />
+                <CommandBar
+                    className={"command-bar"}
+                    items={this._buttons}
+                    overflowMenuProps={
+                        { calloutProps: { className: calloutClassName } as ICalloutProps, onDismiss: calloutOnDismiss, className: "rooster-command-bar-overflow" } as Partial<
+                            IContextualMenuProps
+                        >
```
Comment (pranavgk) [inline packages/roosterjs-react-command-bar/lib/components/RoosterCommandBar.tsx]: nit format `Partial<IContextualMenuProps>`

File: packages/roosterjs-react-emoji/package.json [hunk 3]
```diff
@@ -2,7 +2,7 @@
   "name": "roosterjs-react-emoji",
   "description": "Emoji plugin for roosterjs-react editor",
   "dependencies": {
-    "roosterjs-react-ribbon": ""
+    "roosterjs-react-common": ""
```
[the remark this answers is not in the document]
Comment (adriantran) [reply packages/roosterjs-react-emoji/package.json]: quick answer no

**Pull request merged**